File: app/src/main/java/tn/esprit/wayfinder/models/DestinationVideoModels.kt
package tn.esprit.wayfinder.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class DestinationVideoStatus(
    @SerialName("status") val status: String, // "not_started" | "processing" | "ready" | "failed"
    @SerialName("videoUrl") val videoUrl: String? = null,
    @SerialName("imageCount") val imageCount: Int = 0,
    @SerialName("generatedAt") val generatedAt: String? = null,
    @SerialName("errorMessage") val errorMessage: String? = null
)

@Serializable
data class GenerateVideoResponse(
    @SerialName("status") val status: String,
    @SerialName("userId") val userId: String,
    @SerialName("destination") val destination: String,
    @SerialName("imageCount") val imageCount: Int,
    @SerialName("message") val message: String
)

@Serializable
data class DestinationWithVideoStatus(
    @SerialName("destination") val destination: String,
    @SerialName("videoStatus") val videoStatus: String,
    @SerialName("videoUrl") val videoUrl: String? = null,
    @SerialName("imageCount") val imageCount: Int = 0,
    @SerialName("errorMessage") val errorMessage: String? = null
)

@Serializable
data class UserDestinationsResponse(
    @SerialName("destinations") val destinations: List<DestinationWithVideoStatus>
)

// --- AI Travel Video Models --- //

@Serializable
data class AiVideoStatusResponse(
    @SerialName("available") val available: Boolean,
    @SerialName("suggestions") val suggestions: List<String> = emptyList(),
    @SerialName("message") val message: String? = null
)

@Serializable
data class AiVideoSuggestionsResponse(
    @SerialName("suggestions") val suggestions: List<String>
)

@Serializable
data class AiVideoGenerateRequest(
    @SerialName("prompt") val prompt: String
)

@Serializable
data class AiVideoGenerateResponse(
    @SerialName("success") val success: Boolean,
    @SerialName("message") val message: String? = null,
    @SerialName("data") val data: AiVideoGenerateData? = null
)

@Serializable
data class AiVideoGenerateData(
    @SerialName("predictionId") val predictionId: String,
    @SerialName("status") val status: String,
    @SerialName("originalPrompt") val originalPrompt: String,
    @SerialName("enhancedPrompt") val enhancedPrompt: String,
    @SerialName("estimatedTime") val estimatedTime: String? = null
)

@Serializable
data class AiVideoCheckStatusResponse(
    @SerialName("success") val success: Boolean,
    @SerialName("data") val data: AiVideoStatusData? = null
)

@Serializable
data class AiVideoStatusData(
    @SerialName("predictionId") val predictionId: String,
    @SerialName("status") val status: String,
    @SerialName("videoUrl") val videoUrl: String? = null,
    @SerialName("progress") val progress: Int? = null,
    @SerialName("error") val error: String? = null,
    @SerialName("isComplete") val isComplete: Boolean = false,
    @SerialName("isFailed") val isFailed: Boolean = false
)

@Serializable
data class GenericResponse(
    @SerialName("success") val success: Boolean,
    @SerialName("message") val message: String? = null
)

File: WayFinder/Models/DestinationVideoModels.swift
import Foundation

struct DestinationVideoStatus: Decodable {
    let status: String // "not_started" | "processing" | "ready" | "failed"
    let videoUrl: String?
    let imageCount: Int
    let generatedAt: String?
    let errorMessage: String?
    
    enum CodingKeys: String, CodingKey {
        case status
        case videoUrl
        case imageCount
        case generatedAt
        case errorMessage
    }
}

struct GenerateVideoResponse: Decodable {
    let status: String
    let userId: String
    let destination: String
    let imageCount: Int
    let message: String
    
    enum CodingKeys: String, CodingKey {
        case status
        case userId
        case destination
        case imageCount
        case message
    }
}

struct DestinationWithVideoStatus: Decodable, Identifiable {
    let id: String
    let destination: String
    let videoStatus: String
    let videoUrl: String?
    let imageCount: Int
    let errorMessage: String?
    
    enum CodingKeys: String, CodingKey {
        case destination
        case videoStatus
        case videoUrl
        case imageCount
        case errorMessage
    }
    
    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        destination = try container.decode(String.self, forKey: .destination)
        videoStatus = try container.decode(String.self, forKey: .videoStatus)
        videoUrl = try? container.decode(String.self, forKey: .videoUrl)
        imageCount = try container.decode(Int.self, forKey: .imageCount)
        errorMessage = try? container.decode(String.self, forKey: .errorMessage)
        // Use destination as id for Identifiable
        id = destination
    }
    
    init(destination: String, videoStatus: String, videoUrl: String?, imageCount: Int, errorMessage: String?) {
        self.destination = destination
        self.videoStatus = videoStatus
        self.videoUrl = videoUrl
        self.imageCount = imageCount
        self.errorMessage = errorMessage
        self.id = destination
    }
}

struct UserDestinationsResponse: Decodable {
    let destinations: [DestinationWithVideoStatus]
    
    enum CodingKeys: String, CodingKey {
        case destinations
    }
}

